src/Array2D.luna

```diff
@@ -0,0 +1,235 @@
+        x.forEach indexN:
+            record = x.elemPointer indexN
+            cstringPointer = record.read
+            string = cstringPointer.toText
+            cstringPointer.free
+            modifiedRecord = case ft of
+                FieldReal:
+                    real = parseTextToDouble string
+                    value = ValueReal real
+                    f value
+                _:
+                    value = ValueText string
+                    f value
+            record.write (CString.fromText modifiedRecord.toText)
+        self
+
+    def mapColumn n f:
+        x = Array2DVal (self.getSize.getColumns) (self.getStride.getStrideC) n self.ptr
+        x.forEach indexN:
+            record = x.elemPointer indexN
+            cdoublePointer = record.read
+            modifiedRecord = f cdoublePointer
+            record.write  modifiedRecord
+        self
+
+    def findNA:
+        self.forEach indexN:
+            pointer = self.elemPointer indexN
+            if pointer.read.isNull then (Just indexN) else Nothing
+
+    def map f:
+        self.forEach indexN:
+            pointer = self.elemPointer indexN
+            elem = pointer.read
+            case elem.isNull of
+                True: Pointer None . null
+                False:
+                    value = f elem
+                    elem.free
+                    elem.write value
+
+    def copyColumns listOfColumns:
+        l = listOfColumns.length
+        rows = self.getSize.getRows
+        cols = self.getSize.getColumns
+        strideR = self.getStride.getStrideR
+        strideC = self.getStride.getStrideC
+        columnsArray = Array CInt . fromList (listOfColumns . map CInt.fromInt)
+        new_mat = self._cCall "copy_columns" (Pointer CString) [self.getPtr . toCArg, CSize.fromInt rows . toCArg, CSize.fromInt cols . toCArg, CSize.fromInt strideR . toCArg, CSize.fromInt strideC . toCArg, CInt.fromInt l . toCArg, columnsArray.ptr . toCArg]
+        matManaged = ManagedPointer CString . fromPointer (self._cSym "mat_delete") new_mat
+        Array2DVal (SizeVal rows l) (StrideVal l strideC) 0 matManaged
+
+    def copyRows listOfRows:
+        l = listOfRows.length
+        rows = self.getSize.getRows
+        cols = self.getSize.getColumns
+        strideR = self.getStride.getStrideR
+        strideC = self.getStride.getStrideC
+        rowsArray = Array CInt . fromList (listOfRows . map CInt.fromInt)
+        new_mat = self._cCall "copy_rows" (Pointer CString) [self.getPtr . toCArg, CSize.fromInt rows . toCArg, CSize.fromInt cols . toCArg, CSize.fromInt strideR . toCArg, CSize.fromInt strideC . toCArg, CInt.fromInt l . toCArg, rowsArray.ptr . toCArg]
+        matManaged = ManagedPointer CString . fromPointer (self._cSym "mat_delete") new_mat
+        Array2DVal (SizeVal l cols) self.getStride 0 matManaged
+
+    def cTranspose:
+        rows = self.getSize.getRows
```

Comment:
Yeah so this looks kind of non-lunic :D Before you know it there's `getSizeAbstractFactoryInterfaceMethodAdapter` in our Stdlib :D

You'd save some typing with `SizeVal rows cols = self.size` (modulo the `size => shape` renaming :))